packages/astro/src/server/middleware.ts

```diff
@@ -203,9 +203,70 @@ function addMetaTagToHead(htmlChunk: string, hub: Hub, span?: Span): string {
  *
  * exported for testing
  */
-export function interpolateRouteFromUrlAndParams(rawUrl: string, params: APIContext['params']): string {
-  return Object.entries(params).reduce((interpolateRoute, value) => {
-    const [paramId, paramValue] = value;
-    return interpolateRoute.replace(new RegExp(`(/|-)${paramValue}(/|-|$)`), `$1[${paramId}]$2`);
-  }, rawUrl);
+export function interpolateRouteFromUrlAndParams(
+  rawUrlPathname: string,
```

Comment:
l: Could we add comments here about how such a raw url would look like?

Reply:
sure, added in [2b82b5a](https://github.com/getsentry/sentry-javascript/pull/9815/commits/2b82b5a42d743b178e848fe1710f1993b56f46df)